File: Programação/Back-End PHP/02-poo/04-mundocao/feitos em aula/construtor/Cachorro.php
<?php

    class Cachorro {

        private string $nome;
        private int $idade;
        public string $raca;
        public string $corOlhos = "Preto";
        public string $corPelo = "Amarelo";

        public function __construct(string $nome, string $raca)
        {
            $this->setNome($nome);
            $this->raca = $raca;
        }

        public function getNome() {
            return $this->nome;
        }

        private function setNome(string $nome) { 
            $this->nome = $nome;
        }

        public function getIdade() {
            return $this->idade;
        }

        public function setIdade(int $idade) {
            $this->idade = $idade;
        }

        // Comportamentos / Ações
        public function latir(string $latido){
            echo $latido . PHP_EOL;
        }

        public function comer() {
            echo "Estou comendo..!" . PHP_EOL;
        }

        public function dormir(){
            echo "Estou dormindo com olhos fechados.." . PHP_EOL;
        }

        public function correr(){
            echo "Estou correndo.." . PHP_EOL;
        }

        public function acordar(){
            echo "Estou acordando.." . PHP_EOL;
        }

    }

?>

File: programming/java/02-programacao-orientada-objetos/06-classes-abstratas/exemplo-aula/ExemploClasseAbstrata/src/abstracao/Cachorro.java
package abstracao;

public class Cachorro extends Animal {
    public Cachorro(String nome, int idade, double peso) {
        super(nome, idade, peso);
    }

    @Override
    public void emitirSom() {
        System.out.println("Au Au");
    }

    @Override
    public void comer() {
        System.out.println("Cachorro comendo ração");
    }

    @Override
    public void dormir() {
        System.out.println("Cachorro dormindo");
    }
    
}
